python/pyspark/worker.py

```diff
@@ -1254,17 +1302,14 @@ def read_single_udf(pickleSer, infile, eval_type, runner_conf, udf_index, profil
     if profiler == "perf":
         result_id = read_long(infile)
 
-        if _supports_profiler(eval_type):
-            profiling_func = wrap_perf_profiler(chained_func, result_id)
-        else:
-            profiling_func = chained_func
+        profiling_func = wrap_perf_profiler(chained_func, eval_type, result_id)
 
     elif profiler == "memory":
         result_id = read_long(infile)
-        if _supports_profiler(eval_type) and has_memory_profiler:
-            profiling_func = wrap_memory_profiler(chained_func, result_id)
-        else:
+        if not has_memory_profiler:
```

Comment:
We can probably move this check to `wrap_memory_profiler()` and let it handle the logic. I always think we should handle these logic in one location.

Reply:
Makes sense, done